SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestInitialOnly/SpatialTestInitialOnlyForSpawnActor.cpp

```diff
@@ -103,36 +104,24 @@ void ASpatialTestInitialOnlyForSpawnActor::PrepareTest()
 	});
 
 	AddStep(
-		TEXT("Check changed value."), FWorkerDefinition::Client(1),
-		[this]() -> bool {
-			bool IsReady = false;
-			TArray<AActor*> SpawnActors;
-			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
-			for (AActor* Actor : SpawnActors)
-			{
-				ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
-				if (SpawnActor != nullptr)
-				{
-					IsReady = true;
-				}
-			}
-			return IsReady;
-		},
-		[this]() {
+		TEXT("Check changed value."), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
 			TArray<AActor*> SpawnActors;
 			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ASpatialTestInitialOnlySpawnActor::StaticClass(), SpawnActors);
+			AssertEqual_Int(SpawnActors.Num(), 1, TEXT("There should be exactly one InitialOnly actor in the world."));
 			for (AActor* Actor : SpawnActors)
 			{
 				ASpatialTestInitialOnlySpawnActor* SpawnActor = Cast<ASpatialTestInitialOnlySpawnActor>(Actor);
-				if (SpawnActor != nullptr)
+				if (AssertIsValid(SpawnActor, TEXT("SpawnActor should be valid.")))
 				{
-					AssertTrue(SpawnActor->Int_Initial == 1, TEXT("Check Actor.Int_Initial value."));
-					AssertTrue(SpawnActor->Int_Replicate == 2, TEXT("Check Actor.Int_Replicate value."));
+					RequireEqual_Int(SpawnActor->Int_Initial, 1, TEXT("Check Actor.Int_Initial value."));
+					RequireEqual_Int(SpawnActor->Int_Replicate, 2, TEXT("Check Actor.Int_Replicate value."));
 				}
 			}
 
 			FinishStep();
-		});
+		},
+		10.0f);
```

Comment:
Agree there should be a timelimit when using `Require`, can you add it to the other tests you modified?

Reply:
Ah, definitely should be, can check.

Reply:
Meta-point: would be nice to enforce this, but I guess the easiest way to do so is just to have a non-infinite default step timeout.